course04/homework00/shankarvasudevan/sortedarray/src/sarray.h

```diff
@@ -0,0 +1,45 @@
+#include <array>
+#include <iostream>
+
+class sarray
+{
+public:
+	sarray();
+		
+	static constexpr const std::size_t MaxSize = 10;
+
+	void push_back(int value) 
+	{
+		if (mLength == MaxSize)
+		{
+			throw std::out_of_range("Array is at max capacity");
+		}
+
+		mData[mLength++] = value;
+		for (int i=mLength-1; i > 0 && mData[i] <= mData[i-1]; i--)
+		{
+			int tmp = mData[i];
+			mData[i] = mData[i-1];
+			mData[i-1] = tmp;
+		}
+	}
```

Comment:
A few remarks
  - this is too long for both header file and being defined in class def: please move it to .cc
  - I would stick to std::sort, especialy because you do linear search and shift, I dont think its faster
  - style: spaces: `int i = ...`